src/runtime/linux_host_cpu_count.cpp

```diff
@@ -3,8 +3,27 @@
 extern "C" {
 
 extern long sysconf(int);
+extern int sched_getaffinity(int32_t pid, size_t cpusetsize, unsigned long *cpuset);
 
 WEAK int halide_host_cpu_count() {
-    return sysconf(84);
-}
+    unsigned long count = 0, cpuset[8];  // enough bits for 256 CPUs on 32 bit, or 512 CPUs on 64 bit
+    memset(cpuset, 0, sizeof(cpuset));
+    int rv = sched_getaffinity(0, sizeof(cpuset), cpuset);
+    if (rv == 0) {
+        size_t word, bit;
+        for (word = 0; word < sizeof(cpuset) / sizeof(cpuset[0]); word++) {
+            for (bit = 0; bit < sizeof(cpuset[0]) * 8; bit++) {
```

Comment:
Just use __builtin_popcount here to count the number of set bits

Reply:
Done.